File: robocode2-bot-api-java/src/main/java/net/robocode2/events/TickEvent.java
package net.robocode2.events;

import lombok.Builder;
import lombok.EqualsAndHashCode;
import lombok.ToString;
import lombok.Value;
import net.robocode2.BotState;
import net.robocode2.BulletState;

import java.util.List;

/** Event occurring whenever a new turn in a round has started */
@Value
@EqualsAndHashCode(callSuper = true)
@ToString(callSuper = true)
public class TickEvent extends Event {
  /** Current round number */
  int roundNumber;
  /** Current state of this bot */
  BotState botState;
  /** Current state of the bullets fired by this bot */
  List<BulletState> bulletStates;
  /** Events occurring in the turn relevant for this bot */
  List<Event> events;

  @Builder
  private TickEvent(
      int turnNumber,
      int roundNumber,
      BotState botState,
      List<BulletState> bulletStates,
      List<Event> events) {
    this.turnNumber = turnNumber;
    this.roundNumber = roundNumber;
    this.botState = botState;
    this.bulletStates = bulletStates;
    this.events = events;
  }
}

File: robocode-tankroyale-bot-api-csharp/src/events/TickEvent.cs
using System.Collections.Generic;
using Newtonsoft.Json;
using Robocode.TankRoyale.BotApi;

namespace Robocode.TankRoyale.BotApi
{
  /// <summary>
  /// Event occurring whenever a new turn in a round has started.
  /// </summary>
  public sealed class TickEvent : Event
  {
    /// <summary>Current round number.</summary>
    public int RoundNumber { get; }

    /// <summary>Current state of this bot.</summary>
    public BotState BotState { get; }

    /// <summary>Current state of the bullets fired by this bot.</summary>
    public IEnumerable<BulletState> BulletStates { get; }

    /// <summary>Current state of the bullets fired by this bot.</summary>
    public IEnumerable<Event> Events { get; }

    /// <summary>
    /// Initializes a new instance of the TickEvent class.
    /// </summary>
    /// <param name="turnNumber">Turn number.</param>
    [JsonConstructor]
    public TickEvent(int turnNumber, int roundNumber, BotState botState,
        IEnumerable<BulletState> bulletStates, IEnumerable<Event> events) : base(turnNumber) =>
      (RoundNumber, BotState, BulletStates, Events) =
      (roundNumber, botState, bulletStates, events);
  }
}